.github/workflows/ci-dgo-tests.yml

```diff
@@ -0,0 +1,59 @@
+name: ci-dgo-tests
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+      - ready_for_review
+    branches:
+      - master
+  schedule:
+    - cron: "0 4 * * *" # daily at 4:00
+jobs:
+  dgo-tests:
+    runs-on: [self-hosted, x64] # needed to use docker commands
+    steps:
+      - name: Checkout Dgraph repo
+        uses: actions/checkout@v3
+        with:
+          path: dgraph 
+          repository: dgraph-io/dgraph
+          ref: main
+      - name: Checkout Dgo repo
+        uses: actions/checkout@v3
+        with:
+          path: dgo
+          repository: dgraph-io/dgo
+      - name: Move into Dgraph repo
+        run: cd dgraph 
```

Comment:
from what I know - the CD doesn't hold true outside the sub-task. 

You would want to do this inside every sub-task